src/scene.js

```diff
@@ -270,9 +270,13 @@ Scene.prototype.setZoom = function (zoom) {
         this.removeTilesOutsideZoomRange(below, above);
     }
 
+
     this.last_zoom = this.zoom;
     this.zoom = zoom;
-    this.capped_zoom = Math.min(Math.round(this.zoom), this.tile_source.max_zoom || Math.round(this.zoom));
+
+    this.capped_zoom = Math.min(Math.round(this.zoom), this.findMaxZoom() || Math.round(this.zoom));
```

Comment:
Weill have to revisit max zoom for multiple sources, but this probably makes sense for now.